x-pack/plugins/cases/public/components/actions/tags/use_tags_action.tsx

```diff
@@ -37,16 +38,26 @@ export const useTagsAction = ({ onAction, onActionSuccess, isDisabled }: UseActi
     (tagsSelection: TagsSelectionState) => {
       onAction();
       onFlyoutClosed();
-      const casesToUpdate = selectedCasesToEditTags.map((theCase) => {
+
+      const casesToUpdate = selectedCasesToEditTags.reduce((acc, theCase) => {
         const tags = difference(theCase.tags, tagsSelection.unSelectedTags);
         const uniqueTags = new Set([...tags, ...tagsSelection.selectedTags]);
+        const uniqueTagsAsArray = Array.from(uniqueTags.values());
+
+        if (isEqual(theCase.tags, uniqueTagsAsArray)) {
+          onActionSuccess();
+          return acc;
+        }
 
-        return {
-          tags: Array.from(uniqueTags.values()),
-          id: theCase.id,
-          version: theCase.version,
-        };
-      });
+        return [
+          ...acc,
+          {
+            tags: uniqueTagsAsArray,
+            id: theCase.id,
+            version: theCase.version,
+          },
+        ];
+      }, [] as CaseUpdateRequest[]);
```

Comment:
As discussed with @cnasikas , I still see the error toast as an empty array `casesToUpdate` is getting sent to the backend.

Reply:
Confirmed fixed for the bulk edit of tags. 

Note this same type of issue still occurs with the View case page - would be good to fix this in a follow-up PR:

![edit_case_tags_bug](https://user-images.githubusercontent.com/7405507/202723509-b3cedeb2-623d-44ea-8bbd-3af56f0679cd.gif)

Reply:
Created https://github.com/elastic/kibana/issues/145736 to fix this in the View case page.